### ISSUE TITLE
Using default built serializer within custom serializer

### ISSUE THREAD

**Issue by wemped** [google/built_value.dart#672]
Suppose I'm serializing some JSON for one of my models
```
{  
   "id":1234,
   "metadata":{  
      "names": null,
      "title": null,
   },
   "created_at":"DATE"
}
```
and I have a field (`metadata`) that I'd like to be null if ANY of it's fields are null.

My first stab at this looked something like this
```
abstract class ThingModel
    implements Built<ThingModel, ThingModelBuilder> {

  String get id;
  @nullable
  ThingMetadataModel get metadata;

  factory ThingModel([Function(ThingModelBuilder b) updates]) =
      _$ThingModel;

  ThingModel._();

  static Serializer<ThingModel> get serializer =>
      _$thingModelSerializer;
}

abstract class ThingMetadataModel
    implements Built<ThingMetadataModel, ThingMetadataModelBuilder> {

  String get names;
  String get title;

  factory ThingMetadataModel([Function(ThingMetadataModelBuilder b) updates]) =
      _$ThingMetadataModel;

  ThingMetadataModel._();

  static Serializer<ThingMetadataModel> get serializer =>
      _$thingMetadataModelSerializer;
}
```

With this approach if I receive the JSON above, it will fail when serializing the `ThingModel` (because `names` and `title` are not nullable), and I won't have anything. 
But what I'd really like is to just have a null `ThingMetadataModel` on an otherwise filled out `ThingModel`.

So I thought, I'll just make a custom serializer for `ThingMetadataModel` that is basically a `try/catch` wrapper around the default serializer, returning null in the `catch` case where it couldn't serialize `names` or `title`.

```
class CustomThingMetadataModelSerializer implements StructuredSerializer<ThingMetadataModel> {
  @override
  ThingMetadataModel deserialize(Serializers serializers, Iterable serialized,
      {FullType specifiedType = FullType.unspecified}) {
    try {
      final thingMetadataModel = (_$thingMetadataModelSerializer as StructuredSerializer)
          .deserialize(serializers, serialized);
      return thingMetadataModel;
    } catch (e) {
      return null;
    }
  }

  @override
  Iterable serialize(Serializers serializers, ReviewModel object,
          {FullType specifiedType = FullType.unspecified}) =>
      (_$thingMetadataModelSerializer as StructuredSerializer)
          .serialize(serializers, object);

  @override
  Iterable<Type> get types => _$thingMetadataModelSerializer.types;

  @override
  String get wireName => _$thingMetadataModelSerializer.wireName;
}
```

But as soon as I replace `ThingMetadataModel`'s serializer w/ the custom one
```
@BuiltValueSerializer(custom: true)
  static Serializer<ThingMetadataModel> get serializer =>
      CustomThingMetadataModelSerializer();
```

It get a bunch of errors because the generated 'default' serializer is no longer generated.

Is there a way around this? Or is there a better way to get this sort of behavior?
Basically I'd like to be able to handle nested serialization errors that are thrown.

Thanks!

**Comment by davidmorgan** [comment]
There isn't currently a way to write a custom serializer that wraps the generated one.

What you could do though is write a serializer plugin, like `StandardJsonPlugin`; this could look for the problem cases and fix them.

**Comment by wemped** [comment]
I'm not sure I follow how that would help in this case, would you mind diving a bit deeper? Maybe some pseudo code would help?

**Comment by davidmorgan** [comment]
The plugins can inspect and arbitrarily modify the data before the actual serializers run. So you could check for the case that's going to throw and modify it to match what you want.
